library/core/src/iter/loop_desugar.rs

```diff
@@ -0,0 +1,76 @@
+use crate::iter::IntoIterator as RealIntoIterator;
+use crate::iter::TrustedRandomAccessNoCoerce;
+
+#[unstable(feature = "trusted_random_access", issue = "none")]
+#[doc(hidden)]
+
+pub trait IntoIterator {
+    type IntoIter: Iterator;
+
+    #[unstable(feature = "trusted_random_access", issue = "none")]
+    // #[cfg_attr(not(bootstrap), lang = "loop_desugar")]
+    #[cfg_attr(not(bootstrap), lang = "into_iter")]
```

Comment:
similarly, this into_iter feels iffy -- would be better to call it shim_into_iter or similar.